http://cvs.savannah.gnu.org/viewvc/fastjar/jartool.c?root=fastjar&r1=1.60&r2=1.61&view=patch

2010-06-10  Chris Ball  <cjb@laptop.org>

	* jartool.c (add_file_to_jar): Fix write return value check.

--- a/jartool.c	2010/03/01 15:38:43	1.60
+++ b/jartool.c	2010/06/10 08:46:10	1.61
@@ -1258,7 +1258,7 @@
       exit_on_error("write");
 
   /* write the file name to the zip file */
-  if (1 == write(jfd, fname, file_name_length))
+  if (-1 == write(jfd, fname, file_name_length))
     exit_on_error("write");
 
   if(verbose){
